### ISSUE TITLE
End to End Test 

### ISSUE THREAD

**Issue by OwenPriceSkelly** [Garden-AI/garden#461]
We need a new end-to-end test! We've had the old one switched off since the publishing revamp. 

# E2E test wishlist:
  - runs in github actions 
  - pointed at dev infrastructure (api-dev.thegardens.ai) 
  - should hit happy path and sad path for any backend routes 
    - should exercise these routes "like a user would" via CLI commands rather than the backend client directly 
    - for most backend routes this is just authed/not-authed 
  - should run through the tutorial plus remote inference on the demo endpoint 
  - should use all the base images (except for -all, -wisconsin, -tensorflow)
    - maybe 1 tensorflow image but otherwise they're not really worth it 
    - 

tl;dr "embarrasment-prevention script" we can run before a demo 

# what this does not need to be:
  - doesn't need to test docker builds. ideally we'd be able to pre-cache the layers it'll try to build as part of the flow 
  - anything to do with local data (we're going to be replacing that soon)
